### ISSUE TITLE
Works a few seconds and then becomes Unavailable

### ISSUE THREAD

**Issue by thrawnarn** [mletenay/home-assistant-goodwe-inverter#88]
I just got an update of my ARM of the support. Now it only works for a few seconds then it becomes Unavailable. I'm guessing it only only works when it's getting new data, and then it sets it as Unavailable.
My ARM version is: v13
They also upgraded my DSP-board.

This is when it works:
![inverter_1](https://user-images.githubusercontent.com/5276026/143027448-503bdf97-d067-4261-bbd6-bde121af3fef.png)

a few seconds later:
![inverter_2](https://user-images.githubusercontent.com/5276026/143027523-f3c157a9-ca2b-4336-bee0-a9b49438aea3.png)

This is the history of the inverter entity:
![inverter_3](https://user-images.githubusercontent.com/5276026/143027602-e36044c6-9f02-4965-9223-7c5c466ee02a.png)

**Comment by mletenay** [comment]
This is very likely caused by the unreliability of the network, i.e. not all requests were responded or responded in time.
Try to set the integration log level to debug and see if there are failed requests.
Anyway, there's not much shat can be done on the integration page, only to improve your network stability.

**Comment by TPochobradsky** [comment]
I have a similar problem. What HW components do you use? I have HA on Synology VMM and Microtic router.

**Comment by kolossboss** [comment]
I have the same problem for month now. I‘m not sure whats the problem.
I forced my installer to replace the wifi modul, with the wired Ethernet modul. But same problem.
I have a very reliable unifi system with zero other issues.
I played with the Integration settings like scan interval and Network timeout. Its better, but still sometimes not available.
![image](https://user-images.githubusercontent.com/11015086/151513923-8bc28dea-f252-4b08-b6a3-9864e961d619.jpg)

**Comment by thrawnarn** [comment]
Yeah, I still have the same issue. My network is superstable with unifi products. And I am using ethernet all the way. I used the wireless module before I used the Lan module and I had zero problems with that, so its not a network problem.

**Comment by kolossboss** [comment]
Maybe its a issue with the unifi devices?

**Comment by TPochobradsky** [comment]
I have tried to connect HA with Goodwe through another router and cables without significant changes. :-( 
There are two potential problematics points. The wired Ethernet modul or Instalation HA in VMM on Synology NAS.
Any idea what to tri change?

**Comment by kolossboss** [comment]
I tried the wireless and wired modul. No changes. 
My HA running with HassOS on an Intel NUC and its connected to the same Unifi Switch

**Comment by TPochobradsky** [comment]
My HA running on Synology DS218+ (DSM 7.0.1) -> VMM 2.5.5 -> Home Assistant OS 7.2.
Switch is Mikrotik hAP ac^2.

**Comment by kolossboss** [comment]
I tried the official integration, but its much much worse...
you cant change things like scan interval and network timeout.
Its automatically polling every 10 second. 
But changing this to 30s does not work for me.
The automation in the documentation says, that I need the update `sensor.ppv` But `sensor.ppv` it not availble and `sensor.goodwe_ppv` also did nothing when updating via automation.

**Comment by thrawnarn** [comment]
Ok, I've been looking more into this, because I know that my network isn't bad. And now I think I know whats happend.

Just my backstory again; I had wifi and the integration work flawlessly, but my sems portal didn't work. Because of this I got a LAN module from my supplier. But the LAN module needed a new ARM-firmware to work with the integration. I emailed goodwe who installed the new ARM-firmware, but they also updated my DSP version from v9 to v12. And after this the integration isn't working very good.

My theory is that the DSP-version is answering UDP-calls differently. I think it only answers when something has changed. I wrote a supersimple software to check this manually. It sends the udp request packet with a timeout of 100ms and then retries after 1 second. I'll attach the log for it here. As seen it only sends bytes only
[goodwe.log](https://github.com/mletenay/home-assistant-goodwe-inverter/files/8340579/goodwe.log)
 to the changed value, and when it responds varies heavily.

The log is pretty huge, it's almost 24 minutes (1000 iterations), but I wanted to run it for a longer timespan to show how consistent it is.
